codegen/templates/tchannel_client.tmpl

```diff
@@ -167,24 +167,57 @@ func {{$exportName}}(deps *module.Dependencies) Client {
 		}
 	}
 
-	client := zanzibar.NewTChannelClientContext(
-		channel,
-		deps.Default.ContextLogger,
-		deps.Default.ContextMetrics,
-		deps.Default.ContextExtractor,
-		&zanzibar.TChannelClientOption{
-			ServiceName:          serviceName,
-			ClientID:             "{{$clientID}}",
-			MethodNames:          methodNames,
-			Timeout:              timeout,
-			TimeoutPerAttempt:    timeoutPerAttempt,
-			RoutingKey:           &routingKey,
-			RuleEngine:           re,
-			HeaderPatterns:       headerPatterns,
-			RequestUUIDHeaderKey: requestUUIDHeaderKey,
-			AltChannelMap:        altChannelMap,
-		},
-	)
+	var client *zanzibar.TChannelClient
+
+	retryCountFeaturePresent := deps.Default.Config.ContainsKey("tchannelclients.retryCount.feature.enabled")
+	var retryCountFeatureEnabled bool
+
+	if retryCountFeaturePresent {
+		 retryCountFeatureEnabled = deps.Default.Config.MustGetBoolean("tchannelclients.retryCount.feature.enabled")
+	}
+
+	var maxAttempts int
+	if  retryCountFeatureEnabled && deps.Default.Config.ContainsKey("clients.{{$clientID}}.retryCount") {
```

Comment:
Instead of retryCountFeatureEnabled,Please use deps.Default.Config.ContainsKey("tchannelclients.retryCount.feature.enabled") && deps.Default.Config.MustGetBoolean("tchannelclients.retryCount.feature.enabled")

Reply:
Sure